### PR TITLE
upload mapbc file if it exists

### PR DESCRIPTION
**Pull request by feilin-flexcompute**
If the mapbc file is in the same directory of the ugrid mesh file and their file name's stem is the same, the mapbc file will be  uploaded.

### CODE REVIEW

File: flow360/component/volume_mesh.py [hunk 1]
```diff
@@ -955,6 +955,14 @@ def submit(self, description="", progress_callback=None, compress=True) -> Volum
                 renamed_file_on_remote, self.file_name, progress_callback=progress_callback
             )
 
+        if mesh_parser.is_ugrid() and os.path.isfile(mesh_parser.get_associated_mapbc_file()):
+            remote_mesh_parser = MeshNameParser(renamed_file_on_remote)
+            volume_mesh._webapi._upload_file(
+                remote_mesh_parser.get_associated_mapbc_file(),
+                mesh_parser.get_associated_mapbc_file(),
+                progress_callback=progress_callback,
+            )
```
Comment (benflexcompute) [inline flow360/component/volume_mesh.py:964]: Can you add logging here to notify user when they submit a ugrid but we cannot find its mapbc file?

File: flow360/component/utils.py [hunk 2]
```diff
@@ -535,6 +535,15 @@ def is_valid_surface_mesh(self):
     def is_valid_volume_mesh(self):
         return self.format in [MeshFileFormat.UGRID, MeshFileFormat.CGNS]
 
+    # pylint: disable=missing-function-docstring
+    def get_associated_mapbc_file(self):
```
Comment (benflexcompute) [inline flow360/component/utils.py:539]:
```suggestion
    def get_associated_mapbc_filename(self):
```

**Comment by benflexcompute** [comment]
If user uploads mapbc with ugrid, have you tested if we can do volume_mesh["leftWing"] instead of volume_mesh["1"]?

Does the entire workflow runs without error (for example does case run or not? Is there anywhere in casePipeline/solver where we still use the numbers instead of names?)

**Comment by feilin-flexcompute** [comment]
> If user uploads mapbc with ugrid, have you tested if we can do `volume_mesh["leftWing"]` instead of `volume_mesh["1"]`?
> 
> Does the entire workflow runs without error (for example does case run or not? Is there anywhere in casePipeline/solver where we still use the numbers instead of names?)

yes. The admin link attached did exactly what you said.

**Comment by benflexcompute** [comment]
> > If user uploads mapbc with ugrid, have you tested if we can do `volume_mesh["leftWing"]` instead of `volume_mesh["1"]`?
> > Does the entire workflow runs without error (for example does case run or not? Is there anywhere in casePipeline/solver where we still use the numbers instead of names?)
> 
> yes. The admin link attached did exactly what you said.

It only shows the mesh pipelines works. Is there a case run with it?

**Comment by feilin-flexcompute** [comment]
> > > If user uploads mapbc with ugrid, have you tested if we can do `volume_mesh["leftWing"]` instead of `volume_mesh["1"]`?
> > > Does the entire workflow runs without error (for example does case run or not? Is there anywhere in casePipeline/solver where we still use the numbers instead of names?)
> > 
> > 
> > yes. The admin link attached did exactly what you said.
> 
> It only shows the mesh pipelines works. Is there a case run with it?

the case admin link was attached.

File: flow360/component/utils.py (hunk 2 above)
Comment (feilin-flexcompute) [reply flow360/component/utils.py]: fixed

File: flow360/component/volume_mesh.py (hunk 1 above)
Comment (feilin-flexcompute) [reply flow360/component/volume_mesh.py]: added

**Review by benflexcompute — APPROVED** [review]

**Pull request merged**